components/brave_wallet/browser/json_rpc_service_unittest.cc

```diff
@@ -7488,4 +7509,129 @@ TEST_F(JsonRpcServiceUnitTest, AnkrGetAccountBalances) {
   run_loop_4.Run();
 }
 
+TEST_F(JsonRpcServiceUnitTest, GetSPLTokenProgramByMint) {
+  const std::string tsla_mint_addr =
+      "2inRoG4DuMRRzZxAt913CCdNZCu2eGsDD9kZTrsj2DAZ";
+
+  // Invalid mint or chain ID yields invalid params.
+  TestGetSPLTokenProgramByMint(
+      FROM_HERE, "", mojom::kSolanaMainnet, mojom::SPLTokenProgram::kUnknown,
+      mojom::SolanaProviderError::kInvalidParams,
+      l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+  TestGetSPLTokenProgramByMint(
+      FROM_HERE, tsla_mint_addr, "", mojom::SPLTokenProgram::kUnknown,
+      mojom::SolanaProviderError::kInvalidParams,
+      l10n_util::GetStringUTF8(IDS_WALLET_INVALID_PARAMETERS));
+
+  // Setup registry.
+  const char token_list_json[] = R"(
+    {
+      "2inRoG4DuMRRzZxAt913CCdNZCu2eGsDD9kZTrsj2DAZ": {
+        "name": "Tesla Inc.",
+        "logo": "2inRoG4DuMRRzZxAt913CCdNZCu2eGsDD9kZTrsj2DAZ.png",
+        "erc20": false,
+        "symbol": "TSLA",
+        "decimals": 8,
+        "chainId": "0x65"
+      },
+      "2kMpEJCZL8vEDZe7YPLMCS9Y3WKSAMedXBn7xHPvsWvi": {
+        "name": "SolarMoon",
+        "logo": "2kMpEJCZL8vEDZe7YPLMCS9Y3WKSAMedXBn7xHPvsWvi.png",
+        "erc20": false,
+        "symbol": "MOON",
+        "decimals": 5,
+        "chainId": "0x65",
+        "token2022": true
+      }
+    })";
+
+  auto* registry = BlockchainRegistry::GetInstance();
+  TokenListMap token_list_map;
+  ASSERT_TRUE(
+      ParseTokenList(token_list_json, &token_list_map, mojom::CoinType::SOL));
+  registry->UpdateTokenList(std::move(token_list_map));
+
+  // Setup user asset.
+  auto asset = mojom::BlockchainToken::New(
+      tsla_mint_addr, "Tesla", "tsla.png", false, false, false,
+      mojom::SPLTokenProgram::kToken2022, false, false, "TSLA", 8, true, "", "",
+      mojom::kSolanaMainnet, mojom::CoinType::SOL);
+  ASSERT_TRUE(AddUserAsset(prefs(), asset.Clone()));
+
+  auto asset2 = mojom::BlockchainToken::New(
+      "So11111111111111111111111111111111111111112", "Wrapped SOL", "sol.png",
+      false, false, false, mojom::SPLTokenProgram::kUnknown, false, false,
+      "WSOL", 8, true, "", "", mojom::kSolanaMainnet, mojom::CoinType::SOL);
+  ASSERT_TRUE(AddUserAsset(prefs(), asset2.Clone()));
```

Comment:
can we move this block closer to where we test unknown program in user assets?

Reply:
I would like to keep this one as a part of initializing the state before testing anything.